app/scripts/components/analysis/define/use-custom-aoi.ts

```diff
@@ -0,0 +1,110 @@
+import { Feature, FeatureCollection } from 'geojson';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import shp from 'shpjs';
+
+const extensions = ['geojson', 'json', 'zip'];
+export const acceptExtensions = extensions.map((ext) => `.${ext}`).join(', ');
+
+export interface FileInfo {
+  name: string
+  extension: string
+  type: 'Shapefile' | 'GeoJSON'
+}
+
+function useCustomAoI() {
+  const [fileInfo, setFileInfo] = useState<FileInfo | null>(null);
+  const [uploadFileError, setUploadFileError] = useState<string | null>(null);
+  const [uploadFileWarnings, setUploadFileWarnings] = useState<string[]>([]);
+  const reader = useRef<FileReader>();
+  const [feature, setFeature] = useState<Feature | null>(null);
+  useEffect(() => {
+    reader.current = new FileReader();
+
+    const setError = (error: string) => {
+      setUploadFileError(error);
+      setFeature(null);
+      setUploadFileWarnings([]);
+    };
+
+    const onLoad = async () => {
+      if (!reader.current) return;
+      let geojson: FeatureCollection;
+      if (typeof reader.current.result === 'string') {
+        const rawGeoJSON = reader.current.result;
+        if (!rawGeoJSON) {
+          setError('Error uploading file');
+          return;
+        }
+        try {
+          geojson = JSON.parse(rawGeoJSON as string) as FeatureCollection;
+        } catch (e) {
+          setError('Error uploading file: Invalid JSON');
+          return;
+        }
+      } else {
+        geojson = await shp(reader.current.result);
+      }
+      const feature: Feature = geojson.features[0];
+      if (!feature) {
```

Comment:
Can the type be more specific to avoid the warning: "Unnecessary conditional, value is always falsy"